deployer/src/handlers/mod.rs

```diff
@@ -306,37 +305,34 @@ pub async fn create_service(
     Extension(deployment_manager): Extension<DeploymentManager>,
     Extension(claim): Extension<Claim>,
     Path((project_name, service_name)): Path<(String, String)>,
-    Query(params): Query<HashMap<String, String>>,
-    mut stream: BodyStream,
+    Json(body): Json<DeploymentRequest>,
 ) -> Result<Json<shuttle_common::models::deployment::Response>> {
     let service = persistence.get_or_create_service(&service_name).await?;
     let id = Uuid::new_v4();
 
+    let data = todo!("Base64 decode body.archive");
+
     let deployment = Deployment {
         id,
         service_id: service.id,
         state: State::Queued,
         last_update: Utc::now(),
         address: None,
         is_next: false,
+        git_commit_id: body.git_commit_id,
+        git_commit_msg: body.git_commit_msg,
```

Comment:
Hmm ... might make sense, yet then we maybe also should trim the commit_id, just in case the caller (maybe maliciously) passes a very long string value.

To be super clean, we should use new types like `CommitId` and `CommitMsg` and create associated factories which take a `String` and return a `Result<Self, Error>` in case some validations fail.

I like that approach, but I am not sure if this is in sync with the overall implementation patterns / coding style of Shuttle. Of course we could try to be a shining example.

Reply:
@iulianbarbu, should we just use `commit.summary()` and leave more sphisticated type foo for later?

Reply:
If there was a max limit on commit msg (summary) length, and a user happens to have a very long one, I think they shouldn't be denied to deploy because of that error.
Just trimming it gives shuttle the info it needs.

Reply:
But `summary` is not fallible. It just gets "the short “summary” of the git commit message."

Reply:
It can still be very *not* short :D 
(`commit.summary()` gives the same string)
![image](https://github.com/shuttle-hq/shuttle/assets/54029719/c51c5209-ca83-4122-bf2b-cef231cf32d7)

Reply:
Oh!

Reply:
@hseeberger @jonaro00 , seems like you guys discovered `commit.summary()` is not enough. I think trimming it to 80 chars in addition should do it though.